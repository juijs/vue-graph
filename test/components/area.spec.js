import { mount } from '@vue/test-utils'
import GraphArea from '@/components/area'

describe('area.js', () => {
    it('snapshot', () => {
        const wrapper = mount(GraphArea, {
            props: {
                width: 800,
                height: 400,
                axisFullMode: true,
                shape: 'normal',
                opacity: 0.6,
                borderLine: true,
                labels: [ '1Q', '2Q', '3Q', '4Q' ],
                values: [
                    [ 10, 5, 5, 5 ],
                    [ 40, 10, 10, 10 ],
                    [ 30, 30, 30, 30 ]
                ]
            }
        });

        wrapper.vm.$nextTick(() => {
            expect(wrapper.vm.$el).toMatchSnapshot();
        });
    });
});