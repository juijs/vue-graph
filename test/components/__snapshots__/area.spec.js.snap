// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`area.js snapshot 1`] = `
<div
  style="position: relative; user-select: none; webkit-user-select: none;"
  unselectable="on"
>
  <svg
    buffered-rendering="dynamic"
    height="400"
    style="font-family:arial,Tahoma,verdana,classic;background:#fff"
    width="800"
  >
    <g
      transform="translate(0.5,0.5)"
    >
      <defs>
        <clippath
          id="axis-clip-id-0.0"
        >
          <rect
            height="300"
            width="700"
            x="0"
            y="0"
          />
        </clippath>
        <clippath
          id="axis-clip-rect-id-0"
        >
          <rect
            height="300"
            width="700"
            x="0"
            y="0"
          />
        </clippath>
      </defs>
      <rect
        fill="#fff"
        fill-opacity="0"
        height="300"
        rx="0"
        ry="0"
        stroke="#fff"
        stroke-width="0"
        transform="translate(50,50)"
        width="700"
        x="0"
        y="0"
      />
      <g
        class="grid-fullblock"
        transform="translate(50,350)"
      >
        <g
          transform="translate(0,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            1Q
          </text>
        </g>
        <g
          transform="translate(233.33333333333334,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            2Q
          </text>
        </g>
        <g
          transform="translate(466.6666666666667,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            3Q
          </text>
        </g>
        <g
          transform="translate(700,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            4Q
          </text>
        </g>
        <line
          stroke="#bfbfbf"
          stroke-opacity="1"
          stroke-width="2"
          x1="0"
          x2="700"
          y1="0"
          y2="0"
        />
      </g>
      <g
        transform="translate(50,50)"
      />
      <g
        class="grid-range"
        transform="translate(50,50)"
      >
        <g
          transform="translate(0,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="700"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            80
          </text>
        </g>
        <g
          transform="translate(0,30)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="700"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            72
          </text>
        </g>
        <g
          transform="translate(0,60)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="700"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            64
          </text>
        </g>
        <g
          transform="translate(0,90)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="700"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            56
          </text>
        </g>
        <g
          transform="translate(0,120)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="700"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            48
          </text>
        </g>
        <g
          transform="translate(0,150)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="700"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            40
          </text>
        </g>
        <g
          transform="translate(0,180)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="700"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            32
          </text>
        </g>
        <g
          transform="translate(0,210)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="700"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            24
          </text>
        </g>
        <g
          transform="translate(0,240)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="700"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            16
          </text>
        </g>
        <g
          transform="translate(0,270)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="700"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            8
          </text>
        </g>
        <g
          transform="translate(0,300)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            0
          </text>
        </g>
        <line
          stroke="#bfbfbf"
          stroke-opacity="1"
          stroke-width="2"
          x1="0"
          x2="0"
          y1="0"
          y2="300"
        />
      </g>
      <g
        transform="translate(50,50)"
      />
      <g
        class="grid-panel"
        display="none"
        transform="translate(50,50)"
      >
        <rect
          fill="transparent"
          height="300"
          stroke="transparent"
          width="700"
          x="0"
          y="0"
        />
      </g>
      <g />
      <g
        class="brush-stackarea"
        transform="translate(50,50)"
      >
        <g>
          <path
            cursor="normal"
            d="M0,0 L233.33333333333334,131.25 L466.6666666666667,131.25 L700,131.25"
            fill="transparent"
            stroke="#FFC000"
            stroke-dasharray="none"
            stroke-opacity="0.6"
            stroke-width="2"
            style="pointer-events:stroke"
            x1="0"
            x2="700"
          />
        </g>
        <path
          cursor="normal"
          d="M0,0 L233.33333333333334,131.25 L466.6666666666667,131.25 L700,131.25 L700,300 L0,300 Z"
          fill="#FFC000"
          fill-opacity="0.6"
          stroke="#FFC000"
          stroke-dasharray="none"
          stroke-opacity="0.6"
          stroke-width="0"
          style="pointer-events:stroke"
          x1="0"
          x2="700"
        />
        <g>
          <path
            cursor="normal"
            d="M0,112.5 L233.33333333333334,243.75 L466.6666666666667,243.75 L700,243.75"
            fill="transparent"
            stroke="#7BBAE7"
            stroke-dasharray="none"
            stroke-opacity="0.6"
            stroke-width="2"
            style="pointer-events:stroke"
            x1="0"
            x2="700"
          />
        </g>
        <path
          cursor="normal"
          d="M0,112.5 L233.33333333333334,243.75 L466.6666666666667,243.75 L700,243.75 L700,300 L0,300 Z"
          fill="#7BBAE7"
          fill-opacity="0.6"
          stroke="#7BBAE7"
          stroke-dasharray="none"
          stroke-opacity="0.6"
          stroke-width="0"
          style="pointer-events:stroke"
          x1="0"
          x2="700"
        />
        <g>
          <path
            cursor="normal"
            d="M0,262.5 L233.33333333333334,281.25 L466.6666666666667,281.25 L700,281.25"
            fill="transparent"
            stroke="#7977C2"
            stroke-dasharray="none"
            stroke-opacity="0.6"
            stroke-width="2"
            style="pointer-events:stroke"
            x1="0"
            x2="700"
          />
        </g>
        <path
          cursor="normal"
          d="M0,262.5 L233.33333333333334,281.25 L466.6666666666667,281.25 L700,281.25 L700,300 L0,300 Z"
          fill="#7977C2"
          fill-opacity="0.6"
          stroke="#7977C2"
          stroke-dasharray="none"
          stroke-opacity="0.6"
          stroke-width="0"
          style="pointer-events:stroke"
          x1="0"
          x2="700"
        />
      </g>
      <g />
      <g />
      <g />
    </g>
    <g
      transform="translate(0.5,0.5)"
    />
  </svg>
</div>
`;
